relayer/relays/beefy/on-demand-sync.go

```diff
@@ -72,101 +83,91 @@ func (relay *OnDemandRelay) Start(ctx context.Context) error {
 	}
 
 	gatewayAddress := common.HexToAddress(relay.config.Sink.Contracts.Gateway)
-	gatewayContract, err := contracts.NewGateway(gatewayAddress, relay.ethereumConn.Client())
+	gatewayContractV2, err := contracts.NewGateway(gatewayAddress, relay.ethereumConn.Client())
+	if err != nil {
+		return fmt.Errorf("create gateway client: %w", err)
+	}
+	relay.gatewayContractV2 = gatewayContractV2
+
+	gatewayContractV1, err := contractV1.NewGateway(gatewayAddress, relay.ethereumConn.Client())
 	if err != nil {
 		return fmt.Errorf("create gateway client: %w", err)
 	}
-	relay.gatewayContract = gatewayContract
+	relay.gatewayContractV1 = gatewayContractV1
 
 	relay.tokenBucket.Start(ctx)
 
 	for {
-		sleep(ctx, time.Minute*1)
-		log.Info("Starting check")
-
-		paraNonce, ethNonce, err := relay.queryNonces(ctx)
-		if err != nil {
-			if errors.Is(err, context.Canceled) {
-				return nil
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(time.Second * 30):
+			log.Info("Starting check")
+			err = relay.syncV1(ctx)
+			if err != nil {
```

Comment:
add the check for cancelled context here maybe?

Reply:
Not nessessary, as mentioned https://github.com/Snowfork/snowbridge/pull/1449#discussion_r2103607607